### PR TITLE
Add `clang-tidy` to config and add to CI

### LINKED ISSUE #89: [OSB] Suggestion: Use clang-tidy for static analysis
I was hoping I'd have enough time to actually run it myself, but alas there were too many other things to be getting on with!

`clang-tidy` is an excellent static analysis tool, with a huge library of checks it can run: https://clang.llvm.org/extra/clang-tidy/

Some of the checks are for outright bugs, such as using an object after you've ``std::move``d from it, but it can also make suggestions about readability and performance. It is worth having a look through these and seeing which checks you think would be useful for your own code.

It integrates well with CMake: it can use the generated ``compile_commands.json`` file to find your source files, external headers etc.

In the past I have used it in conjunction with a CI system, but as a first step you could try just running it yourself over the code to see what issues it identifies. (You may need to fix any clang-related compile issues first.)

For reference, here is the CI script I used, including the tests that I used on my own code: https://github.com/BrainsOnBoard/bob_robotics/blob/master/bin/run_clang_tidy_check

### PR DESCRIPTION
**Pull request by alexdewar**
This PR adds a basic `clang-tidy` config file with a minimal set of checks and uses a GitHub Action to use this to lint code in PRs. Note that this Action won't lint existing code -- just the bits introduced by the PR -- so we will still need to run `clang-tidy` locally when we add new checks. I've added the `run-clang-tidy.py` script to help with this; it just lets you run `clang-tidy` in parallel.

I've only enabled a small number of checks for now, because fixing up all the other issues `clang-tidy` can identify (e.g. `const`-ifying methods where possible) would cause a lot of code churn. Nonetheless, it did find a few small issues which I've fixed in this PR and #158. Once this is merged I'll incrementally add checks and fix up the code in separate PRs.

Closes #89.

### CODE REVIEW

File: run-clang-tidy.py [hunk 1]
```diff
@@ -0,0 +1,503 @@
+#!/usr/bin/env python3
```
Comment (jamesturner246) [inline run-clang-tidy.py:1]: On my machine, at least,there's already a `run-clang-tidy` shipped. I can't remember which package. What's the reason to include this here also?

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -88,10 +88,30 @@ jobs:
           cd vcpkg
           ./bootstrap-vcpkg.sh -disableMetrics
 
+      - name: Configure HealthGPS (debug)
```
Comment (jamesturner246) [inline .github/workflows/ci.yml:91]: Might ask to go over this with you in future. Starting to get quite big, and I'm not totally comfortable with the actions files yet.

File: run-clang-tidy.py (hunk 1 above)
Comment (alexdewar) [reply run-clang-tidy.py:1]: Ha, no reason. It turns out I have it too and didn't realise. I'll drop this file then

File: .github/workflows/ci.yml (hunk 2 above)
Comment (alexdewar) [reply .github/workflows/ci.yml:91]: Sure... We could chat tomorrow if you like?

Comment (jamesturner246) [reply .github/workflows/ci.yml:91]: Sure, if you have a moment. Thanks

**Review by jamesturner246 — APPROVED** [review]
I'd say go ahead, once the Python script is dropped. Thanks!

**Comment by alexdewar** [comment]
Thanks for review!

**Pull request merged**